tests/test_misc.py

```diff
@@ -2623,3 +2623,56 @@ def test_force_feerates(node_factory):
         "penalty": 6666,
         "min_acceptable": 1875,
         "max_acceptable": 150000}
+
+
+def test_datastore(node_factory):
+    l1 = node_factory.get_node()
+
+    # Starts empty
+    assert l1.rpc.listdatastore() == {'datastore': []}
+    assert l1.rpc.listdatastore('somekey') == {'datastore': []}
+
+    # Add entries.
+    somedata = b'somedata'.hex()
+    assert l1.rpc.datastore('somekey', somedata) == {'key': 'somekey',
+                                                     'hex': somedata}
+    
+    assert l1.rpc.listdatastore() == {'datastore': [{'key': 'somekey',
+                                                     'hex': somedata}]}
+    assert l1.rpc.listdatastore('somekey') == {'datastore': [{'key': 'somekey',
+                                                              'hex': somedata}]}
+    assert l1.rpc.listdatastore('otherkey') == {'datastore': []}
+
+    otherdata = b'otherdata'.hex()
+    assert l1.rpc.datastore('otherkey', otherdata) == {'key': 'otherkey',
+                                                       'hex': otherdata}
+    
+    assert l1.rpc.listdatastore('somekey') == {'datastore': [{'key': 'somekey',
+                                                              'hex': somedata}]}
+    assert l1.rpc.listdatastore('otherkey') == {'datastore': [{'key': 'otherkey',
+                                                              'hex': otherdata}]}
+    assert l1.rpc.listdatastore('badkey') == {'datastore': []}
+
+    l = l1.rpc.listdatastore()
+    # Order is undefined!
+    assert (l == {'datastore': [{'key': 'somekey', 'hex': somedata},
```

Comment:
I think this is the place where the code check will fail. 

I a little bit difficult to guess what is the correct format of the `{}` from the error message

Reply:
Yes, but it's always better to be *explicit* in tests, than loose.  And it works :)